### PR TITLE
Season refactors and spec with mock data

### PR DESCRIPTION
**Pull request by Sergio-Azcona**
contains updated refactoring and Spec using mock data.
all GREEN!! :)

### CODE REVIEW

File: lib/season.rb [hunk 1]
```diff
@@ -3,158 +3,102 @@
 class Season
   attr_reader :team_data,:game_teams_data
              
-  def initialize(teams_data, game_teams_data)
+  def initialize(teams_data,game_teams_data)
```
Comment (knestler) [inline lib/season.rb:6]: why did you remove the space here?

**Review by knestler — APPROVED** [review]

**Review by DarbySmith — COMMENTED** [review]
Do we need to remove the method logic from the stat_tracker since it is in the seasons class?

File: lib/season.rb (hunk 1 above)
Comment (DarbySmith) [reply lib/season.rb:6]: Not sure either, but my PR removes the initialize method since it is in the parent class.

**Review by DarbySmith — APPROVED** [review]
